Promise.brs

```diff
@@ -53,15 +78,32 @@ sub __nodePromiseResolvedHandler(e as object)
     node = e.getRoSGNode()
     id = node.id
     promise = m[id]
-    promise.context[id + "_callback"](promise.node)
-    promise.complete = true
+
+    isFunc = function (value)
+        valueType = type(value)
+        return (valueType = "roFunction") or (valueType = "Function")
+    end function
+
+    if isFunc(promise.shouldSendCallback) and promise.shouldSendCallback(node) = false then return
+
+    callback = promise.context[id + "_callback"]
+    if isFunc(callback) then
+        if promise.returnSignalFieldValue = true then
```

Comment:
@triwav What is the use case for `returnSignalFieldValue`? I mean, I know it will let the callback know which field was signalling, but since the promise creation already required the user to specify the signal field, when would be a case where it was not known? You cant observe more than one signal field per promise anyway, just just trying to understand the use case.

Reply:
Maybe there is a better name for it. At least in my code I largely just want to know what the actual value is so the first thing I have to do is something like json = node.output. Unless I set additional fields on the node, I don't need to access that all. Make sense?

Reply:
Ah - got it. I have been thinking about this off and on for the past few days and am not sure how I feel about the `returnSignalFieldValue` parameter. I understand the use case now, and agree that often you only care about the signal field value, not the entire object. But the method signatures feel funny to me now, since I often (always?) need to set the `fields` parameter so now I have to provide the `returnSignalFieldValue` parameter as well - it feels like it should be an optional parameter most of the time. I am also not sure that I like the pattern of the promise returning two different types of things depending on how you configured it - just feels kind of funny. The promise syntax was supposed to be a nicer way of writing the normal observer pattern, and in those cases, you still go through the main node. It only saves one short line of code (`json = node.output` like you mentioned) but adds some mental complexity/overhead. But I get that the library should make life easier and this is obviously a pattern that you have ran into often. hmmmmm. I think I would like to think about it a little bit more.
(The rest of the code looks great and I just wanted to say thanks again for putting so much thought into this and for contributing it back.)

Reply:
Thanks Brian. Just one note. If you call `getData()` on the event it's returning the value that changed which is what I use most of the time in a regular observer which is where I thought of the idea. I could acquiesce to switching the order to `fields` then  `returnSignalFieldValue` as false. I actually just modified the signature for __createPromiseFromNode to:
```
function __createPromiseFromNode(node as object, signalField as string, returnSignalFieldValue = false as boolean) as object
```
and moved the option setting into there. I get not wanting to return different things but I do find it easier to use without having to write an extra line each time. It also means your observer and getter will always use the same field so less chance for error there in a refactor.